### ISSUE TITLE
Add apachectl command

### ISSUE THREAD

**Issue by willysr** [certbot/certbot#5332]
## My operating system is (include version):
Slackware 14.2

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
SlackBuild Script (https://slackbuilds.org/repository/14.2/system/letsencrypt/)

## I ran this command and it produced this output:
```
Failed to find executable apache2ctl in PATH: /usr/local/sbin:/usr/local/bin:/sbin:/usr/sbin:/bin:/usr/bin
Certbot doesn't know how to automatically configure the web server on this system. However, it can still get a certificate for you. Please run "certbot certonly" to do so. You'll need to manually configure your web server to use the resulting certificate.
```

## Certbot's behavior differed from what I expected because:
0.19.0 works fine, but when i tested 0.20.0, it couldn't find apache2ctl.
We use apachectl in Slackware. Is it possible to add apachectl in the check list ?

It's almost the same report as https://github.com/certbot/certbot/issues/4282

**Comment by bmw** [comment]
Sorry for the trouble! We landed some major changes in how our Apache plugin handles OS specific differences (such as the name of the ctl command) in 0.20.0. Since you said this worked in 0.19.0, it looks like we had a regression on Slackware.

@joohoi, can you take a look at this?

**Comment by willysr** [comment]
I think other distributions are affected as well
I saw archlinux also uses apachectl instead of apache2ctl
https://www.archlinux.org/packages/extra/x86_64/apache/

**Comment by willysr** [comment]
It seems you have ARC Linux override
https://github.com/certbot/certbot/blob/master/certbot-apache/certbot_apache/override_arch.py

probably the same thing can be added for Slackware?

**Comment by myxor** [comment]
Same goes for Alpine Linux where the executable lies here:

> bash-4.3# which apachectl
/usr/sbin/apachectl

Can i manually override the config options somehow?

**Comment by joohoi** [comment]
> Can i manually override the config options somehow?
Unfortunately it's not currently possible. In 0.20 the distribution based constants were cleaned up a bit to allow specifying this in the future though.

What is confusing me a bit here, is the fact that even prior to 0.20, the `apache2ctl` was supposed to be the default if no overriding distribution fingerprint was found, as can be seen here: https://github.com/certbot/certbot/blob/0.19.x/certbot-apache/certbot_apache/constants.py#L10-L13

The overrides for Slackware and Alpine linux are pretty easy to make, but I'll need to do installations of both to be able to confirm default paths for configuration root, control scripts and their features.

Does Alpine Linux or Slackware have the file `/etc/os-release`  in place, and if they do, what are the contents? This is the file we're opting for for distribution fingerprinting currently.

**Comment by willysr** [comment]
```
cat /etc/os-release 
NAME=Slackware
VERSION="14.2"
ID=slackware
VERSION_ID=14.2
PRETTY_NAME="Slackware 14.2"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:slackware:slackware_linux:14.2"
HOME_URL="http://slackware.com/"
SUPPORT_URL="http://www.linuxquestions.org/questions/slackware-14/"
BUG_REPORT_URL="http://www.linuxquestions.org/questions/slackware-14/"
```

**Comment by willysr** [comment]
I think this configuration can be used for Slackware

```
OS_DEFAULTS = dict(
        server_root="/etc/httpd",
        vhost_root="/etc/httpd/extra",
        vhost_files="httpd-vhosts.conf",
        logs_root="/var/log/httpd",
        version_cmd=['apachectl', '-v'],
        apache_cmd="apachectl",
        restart_cmd=['apachectl', 'graceful'],
        conftest_cmd=['apachectl', 'configtest'],
        enmod=None,
        dismod=None,
        le_vhost_ext="-le-ssl.conf",
        handle_mods=False,
        handle_sites=False,
        challenge_location="/etc/httpd/extra",
        MOD_SSL_CONF_SRC=pkg_resources.resource_filename(
            "certbot_apache", "options-ssl-apache.conf")
)
```

I'm not sure about MOD_SSL_CONF_SRC though. Do you mean the SSL/TLS definition for Apache? If so, then it should be httpd-ssl.conf (located in /etc/httpd/extra/). Also not sure about le_vhost_ext. Can you describe what that field is used for?

Thanks

**Comment by myxor** [comment]
Alpine Linux:
> bash-4.3# cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.6.2
PRETTY_NAME="Alpine Linux v3.6"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"

**Comment by myxor** [comment]
Any progress here or can we anyhow help?

**Comment by joohoi** [comment]
We're currently busy with TLS-SNI change, but if you would like to create a PR for this, it would be most helpful.

**Comment by willysr** [comment]
@joohoi : Since TLS-SNI migration to HTTP-01 is done, is there any time estimation for this issue?

i tried to create override_slack.py and edit the entrypoints.py to add Slackware override, but still it won't detect it. Any clue where i missed it?

Thanks

**Comment by willysr** [comment]
Fixed in 0.27.1.

Thanks @joohoi and @bmw
